trigger/kafkasub/trigger.go

```diff
@@ -119,49 +121,53 @@ func run(t *KafkaSubTrigger) {
 		panic(fmt.Errorf("Failed to create Kafka consumer for reason [%s]", error))
 	}
 	t.kafkaConsumer = &kafkaConsumer
-	thing := make(map[int32]sarama.PartitionConsumer)
-	t.partitionConsumers = &thing
-	flogoLogger.Debugf("Subscribing to topic [%s]", t.kafkaParms.topic)
-	validPartitions, error := kafkaConsumer.Partitions(t.kafkaParms.topic)
-	if error != nil {
-		panic(fmt.Errorf("Failed to get valid partitions from Kafka Consumer for reason [%s].  Aborting subscriber", error))
-	}
-
-	flogoLogger.Debugf("Valid partitions for topic [%s] detected as: [%v]", t.kafkaParms.topic, validPartitions)
-	if t.kafkaParms.partitions == nil { //subscribe to all valid partitions
-		t.kafkaParms.partitions = validPartitions
-		for _, part := range validPartitions {
-			startConsumer(t, part)
+	consumers := make(map[string]sarama.PartitionConsumer)
+	t.partitionConsumers = &consumers
+	for id, handler := range t.kafkaParms.handlers {
+		validPartitions, error := kafkaConsumer.Partitions(handler.topic)
+		if error != nil {
+			panic(fmt.Errorf("Failed to get valid partitions for topic [%s] for reason [%s].  Aborting subscriber",
```

Comment:
Start() in a trigger should not panic, just return an error and the engine will handle it.